corehq/motech/fhir/models.py

```diff
@@ -334,3 +337,103 @@ def deepmerge(a, b):
         return a
     else:
         return b
+
+
+class SmartConfiguration(JsonObject):
+    """
+    The /.well-known/smart-configuration.json response for a domain.
+    http://hl7.org/fhir/smart-app-launch/conformance/index.html#using-well-known
+    """
+    authorization_endpoint = StringProperty(required=True)
+    token_endpoint = StringProperty(required=True)
+    token_endpoint_auth_methods = ListProperty(
+        StringProperty(choices=["client_secret_post", "client_secret_basic"]), exclude_if_none=True
+    )
+    registration_endpoint = StringProperty(exclude_if_none=True)
+    scopes_supported = ListProperty(StringProperty, exclude_if_none=True)
+    response_types_supported = StringProperty(exclude_if_none=True)
+    management_endpoint = StringProperty(exclude_if_none=True)
+    revokation_endpoint = StringProperty(exclude_if_none=True)
+    capabilities = ListProperty(StringProperty, exclude_if_none=True)
+
+
+class _Coding(JsonObject):
```

Comment:
nit: I think a better order for these classes would be
```
_MetadataRest
_RestSecurity
_CodableConcept
_Coding
_Extension
_ExtensionURI
```
the current order made a bit hard to follow the code for review.

(I keep forgetting the name for this flow of code, something we read as a part of clean code book)

Reply:
Yeah I agree that this is a bit confusing, but unfortunately these have to remain in this order because the dependencies flow downwards. (`_CodableConcept` needs `_Coding` to be defined first, etc)